apis/fluent/atrium-api-fluent/src/commonMain/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iteratorExpectations.kt

```diff
@@ -28,3 +26,29 @@ fun <T : Iterator<*>> Expect<T>.toHaveNext(): Expect<T> =
  */
 fun <T : Iterator<*>> Expect<T>.notToHaveNext(): Expect<T> =
     _logicAppend { hasNotNext() }
+
+/**
+ * Expects that `this` expectation (an [Iterator]) has a next element and
+ * returns an [Expect] for the next element.
+ *
+ * @return The newly created [Expect] for the next element.
+ *
+ * @sample ch.tutteli.atrium.api.fluent.en_GB.samples.IteratorExpectationSamples.nextFeature
+ *
+ * @since 1.3.0
+ */
+fun <E, T : Iterator<E>> Expect<T>.next(): Expect<E> =
+    _logic.next().transform()
+
+/**
+ * Expects that `this` expectation (an [Iterator]) has a next element and that
+ * the next element holds all assertions the given [assertionCreator] creates for it.
+ *
+ * @return an [Expect] for the subject of the next expectation.
+ *
+ * @sample ch.tutteli.atrium.api.fluent.en_GB.samples.IteratorExpectationSamples.next
+ *
+ * @since 1.3.0
+ */
+fun <E, T : Iterator<E>> Expect<T>.next(assertionCreator: Expect<E>.() -> Unit): Expect<T> =
```

Comment:
move to iteratorFeatureExtractors.kt (the file does not exist yet, create it please)